fuel-vm/tests/flow.rs

```diff
@@ -416,6 +416,110 @@ fn revert_from_call_immediately_ends_execution() {
     assert_eq!(revert_receipts.len(), 1);
 }
 
+#[test]
+fn nested_call_limit() {
+    let rng = &mut StdRng::seed_from_u64(2322u64);
+
+    let mut client = MemoryClient::default();
+
+    let gas_price = 0;
+    let gas_limit = 1_000_000;
+    let maturity = 0;
+    let height = 0;
+    let params = ConsensusParameters::DEFAULT;
+
+    let salt: Salt = rng.gen();
+    let program: Witness = [op::call(0x10, RegId::ZERO, 0x10, RegId::CGAS), op::ret(RegId::ONE)]
+        .into_iter()
+        .collect::<Vec<u8>>()
+        .into();
+
+    let contract = Contract::from(program.as_ref());
+    let contract_root = contract.root();
+    let state_root = Contract::default_state_root();
+    let contract = contract.id(&salt, &contract_root, &state_root);
+
+    let output = Output::contract_created(contract, state_root);
+
+    let bytecode_witness = 0;
+    let tx = Transaction::create(
+        gas_price,
+        gas_limit,
+        maturity,
+        bytecode_witness,
+        salt,
+        vec![],
+        vec![],
+        vec![output],
+        vec![program],
+    )
+    .into_checked(height, &params, client.gas_costs())
+    .expect("failed to generate checked tx");
+
+    client.deploy(tx);
+
+    let input = Input::contract(rng.gen(), rng.gen(), rng.gen(), rng.gen(), contract);
+    let output = Output::contract(0, rng.gen(), rng.gen());
+
+    let mut script_ops = vec![
+        op::movi(0x10, 0),
+        op::call(0x10, RegId::ZERO, 0x10, RegId::CGAS),
+        op::ret(RegId::ONE),
+    ];
+
+    let script: Vec<u8> = script_ops.clone().into_iter().collect();
```

Comment:
FYI, you can use the GTF opcode now to get the script data length dynamically instead of needing to create 2 versions of the transaction.